modules/core/src/v2/coins/xrp.ts

```diff
@@ -376,40 +376,42 @@ export class Xrp extends BaseCoin {
    * - txHex: hexadecimal representation of transaction
    * @returns {{displayOrder: [string,string,string,string,string], id: *, outputs: Array, changeOutputs: Array}}
    */
-  public explainTransaction(params: ExplainTransactionOptions): TransactionExplanation {
-    let transaction;
-    let txHex;
-    try {
-      transaction = rippleBinaryCodec.decode(params.txHex);
-      txHex = params.txHex;
-    } catch (e) {
+  explainTransaction(params: ExplainTransactionOptions): Bluebird<TransactionExplanation> {
+    return co(function *() {
+      let transaction;
+      let txHex;
       try {
-        transaction = JSON.parse(params.txHex);
-        txHex = rippleBinaryCodec.encode(transaction);
+        transaction = rippleBinaryCodec.decode(params.txHex);
```

Comment:
Why is this change here?

Reply:
git is showing it more complicated than it is... this try/catch matches the one in line 382 of the old version, the next one in line 387 is the same as the one in line 386 of the old version. Only the indentation changed.